language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/ErrorHandleInsideCodeAction.java

```diff
@@ -89,7 +89,8 @@ public int priority() {
         }
 
         edits.add(createVarTextEdits.edits.get(0));
-        edits.addAll(createVarTextEdits.imports);
+        // Add all the import text edits excluding duplicates
+        createVarTextEdits.imports.stream().filter(edits::contains).forEach(edits::add);
```

Comment:
What if we filter the duplicate imports before reaching this point, within `getPossibleTypes`? Otherwise we will have to do this filtering multiple places

Reply:
@nadeeshaan I had a look at the possibility and seems like we cannot do that. Here, we rely on 2 methods to get 
1. create variable text edit -> `CreateVariableOut createVarTextEdits = getCreateVariableTextEdits(range, positionDetails, typeDescriptor.get(), context);` line
2. type guard text edit -> `List<TextEdit> edits = CodeActionUtil.getTypeGuardCodeActionEdits(createVarTextEdits.name, range, unionType, context);` line

And we combine both to come up with the text edits for `create variable and typeguard` action. The above 2 methods return the text edit required for the import statement separately (and may be calling `getPossibleTypes()` twice internally.

Therefore, seems like we have to process this manually as we rely on utils of another 2 atomic code action outputs to provide this code action. WDYT?

Reply:
Okay, then we might need to revisit the implementation later. Since this is acceptable now, let's proceed. +1